### PR TITLE
Example training code over oxford pet dataset

### PR DESCRIPTION
**Pull request by eddyxu**
A Pytorch lighting model to train classification models over Oxford Pet dataset.

### CODE REVIEW

**Comment by eddyxu** [comment]
This PR is built on top of `lei/pytorch` branch.

**Comment by eddyxu** [comment]
Benchmark:


```
time ./train_pet.py -F raw s3://eto-public/datasets/oxford_pet/
9:55.60
time ./train_pet.py -F lance s3://eto-public/datasets/oxford_pet/oxford_pet.lance
4:07.46

```

File: cpp/src/lance/encodings/plain.cc [hunk 1]
```diff
@@ -173,6 +173,15 @@ class PlainDecoderImpl : public Decoder {
     return fmt::format("PlainEncoder({})", type_->ToString());
   }
 
+ protected:
```
Comment (changhiskhan) [inline cpp/src/lance/encodings/plain.cc:176]: just curious why protected instead of private?

File: python/benchmarks/parse_pet.py [hunk 2]
```diff
@@ -78,8 +78,8 @@ def read_metadata(self, check_quality=False) -> pd.DataFrame:
             no_index = pd.Index(names.values).difference(df.filename)
             self._data_quality_issues["missing_index"] = no_index
 
-        # TODO lance doesn't support writing booleans yet
-        with_xmls['segmented'] = with_xmls.segmented.astype(pd.Int8Dtype())
+        with_xmls['segmented'] = with_xmls.segmented.apply(
```
Comment (changhiskhan) [inline python/benchmarks/parse_pet.py:81]: `with_xmls.segments.astype('boolean')` should work (https://pandas.pydata.org/docs/user_guide/boolean.html) no?

**Review by changhiskhan — APPROVED** [review]
couple of nits

File: python/benchmarks/parse_pet.py (hunk 2 above)
Comment (eddyxu) [reply python/benchmarks/parse_pet.py:81]: yup I tried, but it does not work. So the segmented is mixed with string and float NaN.

**Pull request merged**